### ISSUE TITLE
What is the point of the --gguf-model-path in ilab model train?

### ISSUE THREAD

**Issue by maxdebayser** [instructlab/instructlab#2277]
Note: I'm willing to contribute a fix for this behavior which I think is wrong, but I'm opening this issue first to get the opinion of the maintainers on this.

**What is the current experience of ilab you've found issue with?**

I'm testing different combinations of commands for a workshop. Because of slow internet, I would like to avoid having to download tons of model files. So in the `ilab model train` workflow for mac I saw the `--gguf-model-path` flag and thought that it could be used to just re-use the model that is downloaded by default in `ilab model download`. However, there are several gotchas:
- If you pass any `--strategy` flag this setting is silently ignored:
  `model/train.py:467`
```python
    # if macos, preserve that path
    if utils.is_macos_with_m_chip() and not strategy:
```
- If you're not on a Mac the setting is silently ignored
- If you don't pass `--tokenizer-dir` this option is silently ignored:
    `model/train.py:500`
```python
        if tokenizer_dir is not None and gguf_model_path is not None:
            if not local:
```
- Nit: why is it `--tokenizer-dir` and not `--tokenizer-path` like other arguments?
- If you do jump over these hurdles and pass `--gguf-model-path` and `--tokenizer-dir`, you get this error:
```
  File "/Users/m1/clean1/venv/lib/python3.11/site-packages/instructlab/model/train.py", line 544, in train
    for filename in os.listdir(model_dir_local):
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/m1/Library/Application Support/instructlab/checkpoints/instructlab-granite-7b-lab'
```
This raises two questions:

1.  What command should I have run first to make sure that granite or any other model is in the checkpoint directory first?
2. If I need to have the safetensors model in the checkpoint directory, then what's the point of loading the gguf model?

**How do you think the described experience should be?**

Either the gguf model should be converted to safetensors during the `train` command or this command line option should be removed entirely. And if it's not removed, there should be warnings in the cases where the flag is ignored.

**What is the benefit of the change you are proposing?**

Less confusion for the users.

**If applicable, how would backwards-compatability be handled?**

I don't know if it worked in previous version, but it's not working now, so there is no backward-compatibility to preserve.

**Comment by cdoern** [comment]
you are right, this seems to be dead code (the flag itself isnt really used or relevant anymore). We have roadmapped removing this for a while and with #2247 it should get removed.

One should never train off of a GGUF because it has lower performance

**Comment by jaideepr97** [comment]
Thanks for raising this @maxdebayser 
the initial idea for this was in fact allowing users to convert their GGUF models to safetensors, however we found that the conversion was rather lossy and leading to bad models being produced
We haven't gotten the chance to fixing that yet

I agree with @cdoern and you that `--gguf-model-path` should be removed 
my ultimate vision is to default downloading of models to `safetensors` first, which can be converted to gguf when needed for serving etc (this conversion works fine and is not lossy) - this would also achieve the goal of preventing users from downloading the same model twice

At the same time, we are hoping to redesign `ilab model convert` so that it actually functions as described in https://github.com/instructlab/instructlab/issues/2351
I think it would be great to add the gguf -> safetensors conversion path as an option in this scenario, once we have a viable, non-lossy avenue to do this

In the mean time I am including this in a larger epic to revisit a number of flags that need to be deprecated or removed.
Thanks for your patience and understanding!
